packages/dds/tree/src/simple-tree/toMapTree.ts

```diff
@@ -333,16 +324,16 @@ function arrayToMapTree(data: InsertableContent, schema: TreeNodeSchema): Exclus
 
 	const allowedChildTypes = normalizeAllowedTypes(schema.info as ImplicitAllowedTypes);
 
-	const mappedData = arrayToMapTreeFields(data, allowedChildTypes, undefined);
+	const mappedData = Array.from(data, (child) =>
+		arrayChildToMapTree(child, allowedChildTypes),
+	);
 
-	// Array node children are represented as a single field entry denoted with `EmptyKey`
-	const fieldsEntries: [FieldKey, ExclusiveMapTree[]][] =
-		mappedData.length === 0 ? [] : [[EmptyKey, mappedData]];
-	const fields = new Map<FieldKey, ExclusiveMapTree[]>(fieldsEntries);
+	// Array nodes are have a single `EmptyKey` field:
+	const fieldsEntries = mappedData.length === 0 ? [] : ([[EmptyKey, mappedData]] as const);
```

Comment:
when dealing with tuple style arrays and default type inference is too general, `as const` can solve most issues as long as the data is readonly. Worked well here.